sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -64,16 +65,21 @@ export class Ed25519Keypair implements Keypair {
    * @param secretKey secret key byte array
    * @param options: skip secret key validation
    */
-  static fromSecretKey(secretKey: Uint8Array, options?: { skipValidation?: boolean }): Ed25519Keypair {
+  static fromSecretKey(
+    secretKey: Uint8Array,
+    options?: { skipValidation?: boolean }
+  ): Ed25519Keypair {
     const secretKeyLength = secretKey.length;
     if (secretKeyLength != 64) {
```

Comment:
so the underlying nacl library uses `privkey || pubkey` as the secret key. hence the length here is asserting 64. 

if you want to use plain secret key, the `fromSeed` should be used. https://github.com/MystenLabs/sui/pull/6598/files#diff-27ddf765911913455b20bbd8056d4f486650a93122123711e917b9e6a0044a4aR101

cc https://github.com/MystenLabs/sui/issues/6006